src/Compilers/CSharp/Portable/Symbols/Source/ExplicitInterfaceHelpers.cs

```diff
@@ -225,50 +225,10 @@ public static string GetMemberNameWithoutInterfaceName(string fullName)
                 //do a lookup anyway
             }
 
-            var hasParamsParam = implementingMember.HasParamsParameter();
+            var interfaceMembers = explicitInterfaceNamedType.GetMembers(interfaceMemberName);
 
-            // Setting this flag to true does not imply that an interface member has been successfully implemented.
-            // It just indicates that a corresponding interface member has been found (there may still be errors).
-            var foundMatchingMember = false;
-
-            Symbol implementedMember = null;
 
-            foreach (Symbol interfaceMember in explicitInterfaceNamedType.GetMembers(interfaceMemberName))
```

Comment:
>foreach (Symbol interfaceMember in explicitInterfaceNamedType.GetMembers(interfaceMemberName)) [](start = 12, length = 94)

It feels like all the refactorings done in this file aren't necessary/required for correctness. Please revert.

Reply:
@AlekseyTs 
The initial issue was in part caused by the fact that the check for compile time ambiguities in selecting the implemented method was done at a separate point to the actual selection of the implemented method, as I describe in point 2 of this pull requests description.

My original fix remedied this, however this change no longer has any impact due to the change in rules for selecting the implemented method that the LDM suggested.

I still felt that this change is valuable for correctness, so I suggested that we keep it in the pull request in https://github.com/dotnet/roslyn/pull/34584#issuecomment-479645704. @gafter marked that as #Resolved, so I'm assuming that at the least he didn't disagree.

I'm happy to revert this change completely, or to open a new pull request. Unfortunately this change is impossible to unit test now as it does not affect output, but was unit tested by my tests in InheritanceBindingTests.cs before the LDMs decision was made.

Reply:
> I'm happy to revert this change completely, or to open a new pull request. Unfortunately this change is impossible to unit test now as it does not affect output, but was unit tested by my tests in InheritanceBindingTests.cs before the LDMs decision was made.

Please revert the refactoring, keep the change to a necessary minimum. Thanks.

---
In reply to: [273669456](https://github.com/dotnet/roslyn/pull/34584#discussion_r273669456) [](ancestors = 273669456)